lgc/builder/MiscBuilder.cpp

```diff
@@ -99,20 +99,30 @@ Instruction *MiscBuilder::CreateKill(const Twine &instName) {
 //
 // @param instName : Name to give instruction(s)
 Instruction *MiscBuilder::CreateDemoteToHelperInvocation(const Twine &instName) {
+#if !defined(LLVM_HAVE_BRANCH_AMD_GFX)
+#warning[!amd-gfx] DemoteToHelperInvocation replaced by Kill
+  return CreateKill(instName);
+#else
   // Treat a demote as a kill for the purposes of disabling middle-end optimizations.
   auto resUsage = getPipelineState()->getShaderResourceUsage(ShaderStageFragment);
   resUsage->builtInUsage.fs.discard = true;
 
   return CreateIntrinsic(Intrinsic::amdgcn_wqm_demote, {}, getFalse(), nullptr, instName);
+#endif
 }
 
 // =====================================================================================================================
 // Create a helper invocation query. Only allowed in a fragment shader.
 //
 // @param instName : Name to give instruction(s)
 Value *MiscBuilder::CreateIsHelperInvocation(const Twine &instName) {
+#if !defined(LLVM_HAVE_BRANCH_AMD_GFX)
+#warning[!amd-gfx] IsHelperInvocation replaced by "false"
+  return getFalse();
```

Comment:
Use Instrinsic::amdgcn_ps_live might be more appropriate?

Reply:
That's right. By the way, isn't it kind of strange that amdgcn_wqm_helper returns `!IsHelperInvocation`? Kind of a separate topic, but...

Reply:
Off topic, but correct, it was a design bug on my part... I have been thinking about fixing it when it gets upstreamed.